Platest/Controllers/SourceController.cs

```
@@ -46,26 +51,12 @@ public SourceController(ISourceLoadListener listener)
         public void ProcessSourceFileAsync(string fileName)
         {
             _fileName = fileName;
-
-            var worker = new BackgroundWorker();
-            worker.DoWork += worker_DoWork;
-            worker.RunWorkerCompleted += worker_RunWorkerCompleted;
-            worker.RunWorkerAsync(fileName);
-
-
-            //_thread = new Thread(StartProcessing);
-            // _thread.Start();
+            var file = worker_DoWork();
+            StartProcessing(file);
         }
 
-        private void worker_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
+        private SourceFile worker_DoWork()
         {
-            var result = e.Result as SourceFile;
-            _listener.OnSourceLoaded(result);
-        }
-
-        private void worker_DoWork(object sender, DoWorkEventArgs e)
-        {
-            _fileName = e.Argument as string;
             if (_fileName == null)
             {
                 throw new NullReferenceException("Имя файла равно null");
```

Please, throw invalidOperationException instead of NullReferenceException